### PR TITLE
Merge all plan definition list pages

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closses #1019 
Reuse [OpenSRPPlansList](https://github.com/onaio/reveal-frontend/blob/e0167411e1ed1fe28422a6b84f985a9e0ba2d2ca/src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx#L63) components to display all plan from openSrp server.
Refactors: [PlanDefinitionList](https://github.com/onaio/reveal-frontend/blob/master/src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx#L61), [DraftPlans](https://github.com/onaio/reveal-frontend/blob/e0167411e1ed1fe28422a6b84f985a9e0ba2d2ca/src/containers/pages/InterventionPlan/PlanningView/DraftPlans/index.tsx#L16) and [RSAssignmentPlansList](https://github.com/onaio/reveal-frontend/blob/73642980905f1050c892697109df2368448833f7/src/containers/pages/IRS/assignments/index.tsx#L69) components.

### CODE REVIEW

File: src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx [hunk 1]
```diff
@@ -10,21 +10,28 @@ import {
...
+} from '../../../../../../constants';
+import {
+  loadOpenSRPPlans,
+  loadPlansByUserFilter,
+} from '../../../../../../helpers/dataLoading/plans';
 import { getQueryParams } from '../../../../../../helpers/utils';
 import { OpenSRPService } from '../../../../../../services/opensrp';
 
+import { displayError } from '../../../../../../helpers/errors';
+import { makePlansByUserNamesSelector } from '../../../../../../store/ducks/opensrp/planIdsByUser';
```
Comment (p-netm) [inline src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx:26]: is the planIdsByUser reducer registered elsewhere

File: src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx [hunk 2]
```diff
@@ -50,21 +61,46 @@ const defaultBodyRenderer = (componentRender: RenderProp) => {
 };
 
 export const defaultProps: OpenSRPPlanListViewProps = {
+  activePlans: [],
```
Comment (p-netm) [inline src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx:64]: lets please rename this to `planStatuses` or sth like that.

File: src/containers/pages/InterventionPlan/PlanDefinitionList/tests/index.test.tsx [hunk 3]
```diff
@@ -291,32 +305,34 @@ describe('components/InterventionPlan/PlanDefinitionList', () => {
...
     const wrapper = mount(
       <Provider store={store}>
         <Router history={history}>
-          <ConnectedPlanDefinitionList {...props} />
+          <PlanDefinitionList {...props} />
         </Router>
       </Provider>
     );
-    await new Promise(resolve => setImmediate(resolve));
+    await flushPromises();
```
Comment (p-netm) [inline src/containers/pages/InterventionPlan/PlanDefinitionList/tests/index.test.tsx:335]: lets reaudit the tests and wrap wrapper.update  calls due to async calls inside act.

File: src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx (hunk 1 above)
Comment (ciremusyoka) [reply src/containers/pages/InterventionPlan/PlanningView/helpers/OpenSRPPlansList/index.tsx]: Had registered in different pages but have moved it only on this page.

File: src/containers/pages/IRS/assignments/index.tsx [hunk 4]
```diff
@@ -25,229 +15,94 @@ import {
...
-  PlanRecordResponse,
-  PlanStatus,
-} from '../../../../store/ducks/plans';
+  createConnectedOpenSRPPlansList,
+  OpenSRPPlanListViewProps,
+} from '../../InterventionPlan/PlanningView/helpers/OpenSRPPlansList';
+import { draftPlansPageBodyFactory } from '../../InterventionPlan/PlanningView/helpers/utils';
 
 /** register the plan definitions reducer */
 reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
```
Comment (p-netm) [inline src/containers/pages/IRS/assignments/index.tsx:30]: Are we still using this for this page?

File: src/containers/pages/InterventionPlan/PlanDefinitionList/tests/index.test.tsx (hunk 3 above)
Comment (ciremusyoka) [reply src/containers/pages/InterventionPlan/PlanDefinitionList/tests/index.test.tsx:346]: have wrapped tests updating component status with act

**Pull request merged**